Makefile

```diff
@@ -1,14 +1,136 @@
-EXPORTERS := $(wildcard exporter/*/.)
-RECEIVERS := $(wildcard receiver/*/.)
+# More exclusions can be added similar with: -not -path './testbed/*'
+ALL_SRC := $(shell find . -name '*.go' \
+                                -not -path './testbed/*' \
+                                -type f | sort)
 
-.DEFAULT_GOAL := all
+# All source code and documents. Used in spell check.
+ALL_SRC_AND_DOC := $(shell find . \( -name "*.md" -o -name "*.go" -o -name "*.yaml" \) \
+                                -type f | sort)
 
-.PHONY: all $(EXPORTERS) $(RECEIVERS)
-all: $(EXPORTERS) $(RECEIVERS)
-$(EXPORTERS):
-	$(MAKE) -C $@
-$(RECEIVERS):
-	$(MAKE) -C $@
+# ALL_PKGS is used with 'go cover'
+ALL_PKGS := $(shell go list $(sort $(dir $(ALL_SRC))))
+
+GOTEST_OPT?= -race -timeout 30s
+GOTEST_OPT_WITH_COVERAGE = $(GOTEST_OPT) -coverprofile=coverage.txt -covermode=atomic
+GOTEST=go test
+GOFMT=gofmt
+GOIMPORTS=goimports
+GOLINT=golint
+GOVET=go vet
+GOOS=$(shell go env GOOS)
+ADDLICENCESE= addlicense
+MISSPELL=misspell -error
+MISSPELL_CORRECTION=misspell -w
+STATICCHECK=staticcheck
+
+GIT_SHA=$(shell git rev-parse --short HEAD)
+BUILD_INFO_IMPORT_PATH=github.com/open-telemetry/opentelemetry-collector-contrib/internal/version
+BUILD_X1=-X $(BUILD_INFO_IMPORT_PATH).GitHash=$(GIT_SHA)
+ifdef VERSION
+BUILD_X2=-X $(BUILD_INFO_IMPORT_PATH).Version=$(VERSION)
+endif
+BUILD_INFO=-ldflags "${BUILD_X1} ${BUILD_X2}"
+
+all-pkgs:
+	@echo $(ALL_PKGS) | tr ' ' '\n' | sort
+
+all-srcs:
+	@echo $(ALL_SRC) | tr ' ' '\n' | sort
+
+.DEFAULT_GOAL := addlicense-fmt-vet-lint-goimports-misspell-staticcheck-test
+
+.PHONY: all
+all: addlicense-fmt-vet-lint-goimports-misspell-staticcheck-test otelcontribcol
+
+.PHONY: addlicense-fmt-vet-lint-goimports-misspell-staticcheck-test
+addlicense-fmt-vet-lint-goimports-misspell-staticcheck-test: addlicense fmt vet lint goimports misspell staticcheck test
+
+.PHONY: e2e-test
```

Comment:
I suggest to remove from makefile for now and add with testbed.

Reply:
Do you plan to remove this from this PR?

Reply:
I thought I did. Checing

Reply:
Looks like the push go through for some reason.